src/Symfony/Component/Process/Messenger/RunProcessMessageHandler.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Process\Messenger;
+
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Exception\RunProcessFailedException;
+use Symfony\Component\Process\Process;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class RunProcessMessageHandler
+{
+    public function __invoke(RunProcessMessage $message): RunProcessContext
+    {
+        $process = Process::fromShellCommandline(
```

Comment:
I missed this in previous reviews: we shouldn't use fromShellCommandline, it's a pending door to command line injection.
Instead we should define commands as arrays.
We might also support shell command lines, but these should be 2nd class.

Reply:
Good call.